app/WebParsing/ParsecCombinators.hs

```diff
@@ -100,16 +94,11 @@ parseUntil parser = do
     parsed <- P.manyTill P.anyChar (P.try parser)
     return $ T.pack parsed
 
-splitPrereqText :: Parser [T.Text]
-splitPrereqText = do
-    P.manyTill (P.try parseNotes <|> P.try parseNoteLine <|>
-        P.try parseCategory <|> parseUntil P.eof) P.eof
-
 parseCategory :: Parser T.Text
-parseCategory = do
-    left <- parseUpToSeparator
-    _ <- P.anyChar
-    return left
+parseCategory = parseNumberedLine <|> parseLine
+
+parseLine :: Parser T.Text
+parseLine = parseUpToSeparator <|> parseUntil P.eof
```

Comment:
Given that each req string is a single line from the HTML, it actually shouldn't be necessary to use `parseUpToSeparator`, as there shouldn't be any `\r` or `\n` characters in the strings being passed to `parseCategory`.

This can help simplify `parseCategory`, by saying that the only "choice" the parser makes is that `P.spaces >> p.digit >> text "." >> P.space >> P.spaces` is *optional*, and then after that read until the end of the line.